subsys/bluetooth/host/gatt.c

```diff
@@ -1361,16 +1363,15 @@ static int gatt_notify(struct bt_conn *conn, u16_t handle,
 	struct net_buf *buf;
 	struct bt_att_notify *nfy;
 
-#if defined(CONFIG_BT_GATT_ENFORCE_CHANGE_UNAWARE)
 	/* BLUETOOTH CORE SPECIFICATION Version 5.1 | Vol 3, Part G page 2350:
 	 * Except for the Handle Value indication, the  server shall not send
 	 * notifications and indications to such a client until it becomes
 	 * change-aware.
 	 */
-	if (!bt_gatt_change_aware(conn, false)) {
+	if (IS_ENABLED(CONFIG_BT_GATT_ENFORCE_CHANGE_UNAWARE) &&
+	    !bt_gatt_change_aware(conn, false)) {
```

Comment:
Same here with `bt_gatt_change_aware()` - this change doesn't do much since the function has an #ifdef inside it which makes it evaluate to a constant `true` if the feature isn't enabled. Did you try removing that #ifdef. There are probably some static variables that will need to have their #ifdefs removed as well for this all to still compile. Anyway, I suppose it makes sense to hold off any changes to service changed or caching until the other pending PR for that feature has been merged.

Reply:
Same thing, I could not remove inner #ifdef because of dependecies to other #ifdefs.